src/python/WMCore/ReqMgr/Service/Request.py

```diff
@@ -426,6 +426,11 @@ def _handleAssignmentStateTransition(self, workload, request_args, dn):
         # by default, it contains all unmerged LFNs (used by sites to protect the unmerged area)
         request_args['OutputModulesLFNBases'] = workload.listAllOutputModulesLFNBases()
 
+        # hacky way to update MaxRSS, if MaxRSS is not set but Memory is set Memory becomes MaxRSS value.
+        # that is how it sets in WMTask setJobResourceInformation method
+        if request_args.get('Memory') and request_args.get('MaxRSS') is not None:
```

Comment:
Honestly, I don't like the way it looks. And your comment made me like it even less :)
I think it can cause more harm then good for cases where we have Memory or MaxRSS also set inside the tasks/steps.

I think it would be better to have some sort of helper function in the StdBase such that we can nicely build the final MaxRSS that goes into the workload_cache (json). The other changes look good.

Reply:
It won't be that bad as it looks. If Memory is int MaxRSS will be int If Memory is dict, MaxRSS will be dict. Getter function would be a little complicated, since it goes through task and returns MaxRSS depending on whether it has the same value or different value by task. (In that case construct the dictionary. I think if we provide getter for this, we need to provide all the getters the same way if it has different values for task.)
I will check how we are doing it.